src/common/Model/Config/AttributeConfig.php

```diff
@@ -263,4 +303,18 @@ public function getSiteVariantAttributes(): array
         }
         return $this->siteVariantAttributes;
     }
+
+    /**
+     * Get mapping of attribute names between Magento and Fredhopper
+     *
+     * @return array
+     */
+    public function getAttributeNameMapping(): array
+    {
+        if (!isset($this->attributeNameMapping)) {
+            $configValue = $this->scopeConfig->getValue(self::XML_PATH_ATTRIBUTE_MAPPING);
+            $this->attributeNameMapping = $this->json->unserialize($configValue ?? '[]') ?? [];
```

Comment:
```suggestion
$this->attributeNameMapping = $this->json->unserialize($configValue ?? '[]');
```
json->unserialize is either going to throw an error if `$configValue` is malformed, or return an empty array if `$configValue` is falsey. No need for a null coalesc right?

Reply:
Can't pass null to it - would produce an `InvalidArgumentException`

```
public function unserialize($string)
{
    if ($string === null) {
        throw new \InvalidArgumentException(
            'Unable to unserialize value. Error: Parameter must be a string type, null given.'
        );
    }
    $result = json_decode($string, true);

    if (json_last_error() !== JSON_ERROR_NONE) {
        throw new \InvalidArgumentException("Unable to unserialize value. Error: " . json_last_error_msg());
    }
    return $result;
}
```

Reply:
The suggestion doesn't say to pass null to `unserialize`, it says you don't need the 2nd coalesc if serialize was to return an empty array.

Reply:
Ah - yep. Missed that

Reply:
Ah OK, no you've sort of got a point here, it's necessary at the moment because you're only using `??`. So here's a full suggestion:

```suggestion
        $this->attributeNameMapping = $this->json->unserialize($configValue ?: '[]');
```

Reply:
For context, because:
```
php > var_dump(json_decode('', true));
NULL
```